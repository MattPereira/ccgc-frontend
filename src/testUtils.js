import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "happy-gilmore",
  firstName: "Happy",
  lastName: "Gilmore",
  email: "happy@gmail.com",
  isAdmin: true,
  rounds: [],
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
