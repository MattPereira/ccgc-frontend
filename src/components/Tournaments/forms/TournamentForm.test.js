import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "../../testUtils";
import TournamentForm from "./TournamentForm";

const courseHandles = ["pebble-beach", "roddy-ranch", "rooster-run"];

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <TournamentForm courseHandles={courseHandles} />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <TournamentForm courseHandles={courseHandles} />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays form inputs and login button", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <TournamentForm courseHandles={courseHandles} />
      </UserProvider>
    </MemoryRouter>
  );

  const dateInput = screen.getByLabelText("Date");
  expect(dateInput).toBeInTheDocument();

  const courseInput = screen.getByLabelText("Golf Course");
  expect(courseInput).toBeInTheDocument();

  const tourYearInput = screen.getByLabelText("Tour Year");
  expect(tourYearInput).toBeInTheDocument();

  //only one button in the component
  const submitBtn = screen.getByRole("button");
  expect(submitBtn).toBeInTheDocument();
});
