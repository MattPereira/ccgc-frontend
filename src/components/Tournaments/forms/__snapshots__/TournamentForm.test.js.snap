// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-sm-10 col-md-8 col-lg-6"
    >
      <div
        class="card"
      >
        <div
          class="display-6 text-center bg-primary py-3 text-white card-title h5"
        >
          New Tournament
        </div>
        <div
          class="card-body"
        >
          <form
            class=""
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="date"
              >
                Date
              </label>
              <input
                class="form-control"
                id="date"
                name="date"
                required=""
                type="date"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="courseHandle"
              >
                Golf Course
              </label>
              <select
                class="form-select"
                id="courseHandle"
                name="courseHandle"
                required=""
                type="select"
              >
                <option
                  value="pebble-beach"
                >
                  pebble-beach
                </option>
                <option
                  value="roddy-ranch"
                >
                  roddy-ranch
                </option>
                <option
                  value="rooster-run"
                >
                  rooster-run
                </option>
              </select>
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="tourYears"
              >
                Tour Year
              </label>
              <select
                class="form-select"
                id="tourYears"
                name="tourYears"
                required=""
                type="select"
              >
                <option>
                  2021-22
                </option>
                <option>
                  2022-23
                </option>
                <option>
                  2023-24
                </option>
                <option>
                  2024-25
                </option>
                <option>
                  2025-26
                </option>
              </select>
            </div>
            <div
              class="row justify-content-end"
            >
              <div
                class="col-auto"
              >
                <button
                  class="btn btn-primary btn-block px-4"
                >
                  Submit
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
